### PR TITLE
feat: add new quickstart guide

### PR DESCRIPTION
**Pull request by zmitchell**

### CODE REVIEW

**Review by stahnma — APPROVED** [review]

**Review by bryanhonof — APPROVED** [review]

File: docs/index.md [hunk 1]
```diff
@@ -6,328 +6,609 @@ description: Get started with creating Flox environments.
...
 ```
-✅ Authentication complete
-✅ Logged in as <user>
-✅ flox_project successfully pushed to FloxHub
 
-Use 'flox pull <user>/flox_project' to get this environment in any other location.
+When you activate the environment you can see that `flox [my_project]` is
+prepended to your prompt so that you can see that you're inside the
+environment.
+If you have more than one environment activate (you can stack them!) this prompt
```
Comment (jennymahmoudi) [inline docs/index.md:442]: typo: activated

File: docs/index.md [hunk 2]
```diff
@@ -6,328 +6,609 @@ description: Get started with creating Flox environments.
...
+will show a list of the active environments.
+Whether to show this prompt is configurable, see [`flox config`][config]
```
Comment (jennymahmoudi) [inline docs/index.md:444]: This feels like a good time to mention what to do if you don't see the prompt (bc of zsh theme, etc). Perhaps?

> If you don't see the prompt change, <do X>. If you want to disable the prompt, see  [`flox config'][config].

If there isn't a simple thing they can do, can we just say "If you use custom zsh themes, your prompt may not change."

File: docs/index.md [hunk 3]
```diff
@@ -6,328 +6,609 @@ description: Get started with creating Flox environments.
...
-1. Activate the environment with [`flox activate`][activate]
-1. Make requests with `httpie`, inspect responses with `fx` and `jq`.
-1. Push the environment to FloxHub with [`flox push`][push]
-1. Delete the local environment with [`flox delete`][delete]
-1. Activate the pushed environment with [`flox activate --remote`][activate]
+1. Install Flox.
+2. Create an environment with `flox init`.
+3. Find your dependencies with `flox search` and `flox show`.
+4. Install dependencies with `flox install`
+5. Add environment variables, shell hooks, and services with `flox edit`.
```
Comment (jennymahmoudi) [inline docs/index.md:25]: YES!!!

**Review by ghudgins — APPROVED** [review]
overall LGTM.

non-blocking feedback: I get why we're doing it but there is a bit of a flow oddity to always switch back to installing go from the language specific initial section. but yeah, it's probably too much to maintain all those paths so I totally get it.

**Comment by zmitchell** [comment]
Yeah I chose to go that route because the portions that I've done in a language specific way have actual differences based on the language whereas the Go-only parts are mostly generic. As a first pass I think this is fine and I don't feel like it disrupts the flow too much.

**Pull request merged**